src/state/stake/hooks.ts

```diff
@@ -61,13 +65,78 @@ export interface StakingInfo {
   readonly rewardTokens: Token[]
 }
 
+export const useMultiRewardPool = (): MultiRewardPool[] => {
+  const library = useProvider()
+  const farmSummaries = useFarmRegistry()
+
+  const [multiRewardPools, setMultiRewardPools] = React.useState<MultiRewardPool[]>([])
+
+  const call = React.useCallback(async () => {
+    const multiRwdPools: MultiRewardPool[] = []
+
+    await Promise.all(
+      farmSummaries.map(async (fs) => {
+        let poolContract = new Contract(fs.stakingAddress, DUAL_REWARDS_ABI, getProviderOrSigner(library) as any)
+        const rewardsToken = []
+        const externalStakingRwdAddresses = []
+
+        // the first reward token at the top level
+        rewardsToken.push(await poolContract.rewardsToken())
+
+        // last time the contract was updated - set isActive to false if it has been longer than 2 months
+        const periodFinish = await poolContract.periodFinish()
+        const isActive = Math.floor(Date.now() / 1000) - periodFinish > 5259492 ? false : true
+
+        let baseContractFound = false
+        // recursivley find underlying and base pool contracts
+        while (!baseContractFound) {
+          try {
+            const externalStakingRewardAddr = await poolContract.externalStakingRewards()
+            externalStakingRwdAddresses.push(externalStakingRewardAddr)
+            poolContract = new Contract(
+              externalStakingRewardAddr,
+              DUAL_REWARDS_ABI,
+              getProviderOrSigner(library) as any
+            )
+            rewardsToken.push(await poolContract.rewardsToken())
+          } catch (e) {
+            //set true when externalStakingRewards() throws an error
+            baseContractFound = true
```

Comment:
I would check for the specific error we expect to see. If there's an unexpected error, we should log it `console.error`